test/gateway/eip20_gateway/confirm_redeem_intent.js

```diff
@@ -0,0 +1,503 @@
+    );
+
+    let blockHeight = new BN(redeemRequest.blockNumber, 16);
+    assert.strictEqual(
+      blockHeight.eq(eventData._blockHeight),
+      true,
+      `Block height from event must be equal to ${blockHeight.toString(10)}.`,
+    );
+
+    assert.strictEqual(
+      eventData._hashLock,
+      expectedEventValues._hashLock,
+      `Hash lock from event must be equal to ${expectedEventValues._hashLock}.`,
+    );
+
+  }
+
+  beforeEach(async function () {
+
+      valueTokenAddress = accounts[0];
+      mockToken = await MockToken.new({from: accounts[0]});
+      baseToken = await MockToken.new({from: accounts[0]});
+
+      redeemRequest = TestDataJSON.gateway.confirm_redeem_intent.params;
+      owner = accounts[4];
+      worker = accounts[8];
+
+      mockOrganization = await MockOrganization.new(owner, worker);
+
+      utilityToken = await Token.new(
+        accounts[3],
+        "",
+        "",
+        18,
+        owner,
+        {from: accounts[0]},
+      );
+
+      coreAddress = accounts[3];
+      burnerAddress = accounts[6];
+      bountyAmount = new BN(100);
+      gatewayAddress = testData;
+      dummyStateRootProvider = accounts[2];
+      eip20Gateway = await EIP20Gateway.new(
+        mockToken.address,
+        baseToken.address,
+        dummyStateRootProvider,
+        bountyAmount,
+        mockOrganization.address,
+        burnerAddress,
+      );
+
+      gatewayUtils = new GatewayUtils(eip20Gateway, mockToken, baseToken);
+      await eip20Gateway.activateGateway(TestDataJSON.contracts.coGateway, {from: owner});
+      await eip20Gateway.setStorageRoot(
+        new BN(redeemRequest.blockNumber, 16),
+        redeemRequest.storageRoot,
+      );
+
+    }
+  );
+
+  it('should pass when all the params are valid', async function () {
+
+      let result = await eip20Gateway.confirmRedeemIntent.call(
+        redeemRequest.redeemer,
+        new BN(redeemRequest.nonce, 16),
+        redeemRequest.beneficiary,
+        new BN(redeemRequest.amount, 16),
+        new BN(redeemRequest.gasPrice, 16),
+        new BN(redeemRequest.gasLimit, 16),
+        new BN(redeemRequest.blockNumber, 16),
+        redeemRequest.hashLock,
+        redeemRequest.storageProof,
+      );
+
+      let messageHash = TestDataJSON.gateway.confirm_redeem_intent.return_value.returned_value.messageHash_;
+      assert.strictEqual(
+        result,
+        messageHash,
+        `Message hash from event must be equal to ${messageHash}.`,
+      );
+
+      await eip20Gateway.confirmRedeemIntent(
+        redeemRequest.redeemer,
+        new BN(redeemRequest.nonce, 16),
+        redeemRequest.beneficiary,
+        new BN(redeemRequest.amount, 16),
+        new BN(redeemRequest.gasPrice, 16),
+        new BN(redeemRequest.gasLimit, 16),
+        new BN(redeemRequest.blockNumber, 16),
+        redeemRequest.hashLock,
+        redeemRequest.storageProof,
+      );
+
+      let message = await eip20Gateway.messages(messageHash);
```

Comment:
Why are you assert message? Should the return value and events assertion will be enough to test? 
@sarvesh-ost what are your thoughts?

Reply:
Agree, it seems there is no need for asserting a message.